Submitted by: Ryan Oliver <ryan.oliver@pha.com.au>
Date: 20050503
Initial Package Version: 4.0.0
Description:
  Removes standard_exec_prefix_2 from library search path when cross-compiling.
  Removes both standard_exec_prefix_1 and standard_exec_prefix_2 from the gcc
  executable search path when cross-compiling.

  ie: if cross_compile = 1 in the specs file, do not
      - add /usr/lib/gcc/${TARGET}/${GCC_VER} to the library search path
      - add /usr/lib/gcc/${TARGET}/${GCC_VER} or
            /usr/libexec/gcc/${TARGET}/${GCC_VER}
        to the executable search path


--- gcc-4.0.0/gcc/gcc.c-orig	2005-04-16 11:38:45.000000000 +1000
+++ gcc-4.0.0/gcc/gcc.c	2005-05-03 22:18:10.000000000 +1000
@@ -3797,16 +3797,22 @@
 	      PREFIX_PRIORITY_LAST, 2, 0);
   add_prefix (&exec_prefixes, standard_exec_prefix, "BINUTILS",
 	      PREFIX_PRIORITY_LAST, 2, 0);
-  add_prefix (&exec_prefixes, standard_exec_prefix_1, "BINUTILS",
+  if (*cross_compile == '0')
+    {
+      add_prefix (&exec_prefixes, standard_exec_prefix_1, "BINUTILS",
 	      PREFIX_PRIORITY_LAST, 2, 0);
-  add_prefix (&exec_prefixes, standard_exec_prefix_2, "BINUTILS",
+      add_prefix (&exec_prefixes, standard_exec_prefix_2, "BINUTILS",
 	      PREFIX_PRIORITY_LAST, 2, 0);
+    }
 #endif
 
   add_prefix (&startfile_prefixes, standard_exec_prefix, "BINUTILS",
 	      PREFIX_PRIORITY_LAST, 1, 0);
-  add_prefix (&startfile_prefixes, standard_exec_prefix_2, "BINUTILS",
+  if (*cross_compile == '0')
+    {
+      add_prefix (&startfile_prefixes, standard_exec_prefix_2, "BINUTILS",
 	      PREFIX_PRIORITY_LAST, 1, 0);
+    }
 
   tooldir_prefix = concat (tooldir_base_prefix, spec_machine,
 			   dir_separator_str, NULL);
